### ISSUE TITLE
unreachable: false positive for isinstance

### ISSUE THREAD

**Issue by tyralla** [python/mypy#9907]
I have come across the problem when working with runtime checkable Protocols and overloading. Still, I can simplify it to the following example, where Mypy (version 0.79) reports a wrong "unreachable" error and thus misses the real bug in the last line:

```
from abc import abstractmethod
from typing_extensions import Literal


class A:
    f: Literal[0]


class B:
    @property
    @abstractmethod
    def f(self) -> Literal[0, 1]:
        ...

    def t(self) -> None:
        if isinstance(self, A):  # error: Subclass of "B" and "A" cannot exist: would have incompatible method signatures  [unreachable]
            self.f = 1  # error: Statement is unreachable  [unreachable]
```

On the other hand, when checking multiple inheritance, Mypy correctly detects that the definition of class `C` is  type-safe but the definition of class `D` is not:

```
class C(A, B):
    ...


class D(B, A):  # error: Definition of "f" in base class "B" is incompatible with definition in base class "A"  [misc]
    ...
```

So method `TypeChecker.check_multiple_inheritance` could eventually help to fix method `TypeChecker.find_isinstance_check_helper`?

As mentioned above, the problem would be the same if class `A` were a runtime checkable protocol.

**Comment by gvanrossum** [comment]
Do you want to send us a PR with a fix?

**Comment by tyralla** [comment]
Yes, I will try it at the weekend.

**Comment by tyralla** [comment]
@gvanrossum Before we forget: pull request #9909 has been waiting to be reviewed for almost two months now.
